src/lighteval/utils/utils.py

```diff
@@ -205,21 +214,32 @@ def boolstring_to_bool(x: Union[str, bool, int]) -> Union[bool, None]:
     raise ValueError(f"You tried to convert {x} to a boolean but it's not possible.")
 
 
-def download_dataset_worker(args):
+def download_dataset_worker(
+    dataset_path: str,
+    dataset_config_name: str,
+    trust_dataset: bool,
+    dataset_filter: Callable[[dict], bool] | None = None,
+    revision: str | None = None,
+) -> DatasetDict:
     """
     Worker function to download a dataset from the HuggingFace Hub.
     Used for parallel dataset loading.
     """
-    dataset_path, dataset_config_name, trust_dataset = args
     dataset = load_dataset(
         path=dataset_path,
         name=dataset_config_name,
         data_dir=None,
         cache_dir=None,
         download_mode=None,
         trust_remote_code=trust_dataset,
+        revision=revision,
     )
-    return dataset
+
+    if dataset_filter is not None:
+        dataset = dataset.filter(dataset_filter)
+
+    # It returns DatasetDict because we don't specify a split
+    return dataset  # type: ignore
```

Comment:
why do we ignore the type here ?

Reply:
Because it doesn't have correct type
load_dataset returns `Dataset | DatasetDict | IterableDataset | IterableDatasetDict`, based on the input args and afaik there is unspecified contract that if the we don't provide `streaming` and `split` arg we get `DatasetDict`. However there is no way to achieve this on typings level, so I just ignore this error.

If the question was why I put there the `type: ignore` it's because even tho we don't have a typechecker in the quality checks, I do have it on in my vscode (pyright) and it shows red everywhere where there is a typing problem.